app/js/app/services/session.js

```diff
@@ -0,0 +1,342 @@
+          iStorage.set('user_id', data.id);
+          iStorage.set('is_registration_complete', data.is_registration_complete);
+        }
+        session.token = data.token;
+        session.user_id = data.id;
+        session.is_registration_complete = data.is_registration_complete;
+        deferred.resolve(data);
+      }).error(function (data, status) {
+        deferred.reject(status);
+      });
+      return deferred.promise;
+    },
+
+    facebookLogin: function (params) {
+      return $http({
+        method: 'POST',
+        url: serverConfig.url + '/api/secure/facebook/login',
+        data: angular.element.param(params)
+      }).then(function (response) {
+        iStorage.set('token', response.data.token);
+        iStorage.set('user_id', response.data.id);
+        iStorage.set('is_registration_complete', response.data.is_registration_complete);
+        session.token = response.data.token;
+        session.user_id = response.data.id;
+        session.is_registration_complete = response.data.is_registration_complete;
+        return response;
+      });
+    },
+
+    registration: function (data) {
+      return $http({
+        method: 'POST',
+        url: serverConfig.url + '/api/secure/registration',
+        data: angular.element.param(data)
+      }).then(function (response) {
+        session.token = response.data.token;
+        session.user_id = response.data.id;
+        session.is_registration_complete = response.data.is_registration_complete;
+
+        if (data.facebook_id) {
+          iStorage.set('token', response.data.token);
+          iStorage.set('user_id', response.data.id);
+          iStorage.set('is_registration_complete', response.data.is_registration_complete);
+        }
+
+        return response;
+      });
+    },
+
+    forgotPassword: function (email) {
+      var data = {
+        email: email
+      };
+      return $http.post(serverConfig.url + '/api/secure/forgot-password', angular.element.param(data), {
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded'
+        }
+      });
+    },
+
+    logout: function () {
+      iStorage.set('token', '');
+      iStorage.set('user_id', '');
+      iStorage.set('is_registration_complete', '');
+      session.token = null;
+      session.user_id = null;
+      session.is_registration_complete = null;
+    }
+  };
+
+  session.getToken = function () {
+    return session.token;
+  };
+
+  session.token = iStorage.get('token');
+  session.user_id = iStorage.get('user_id');
+  session.is_registration_complete = iStorage.get('is_registration_complete');
+
+  if (session.token) {
+    $http.defaults.headers.common.Token = session.token;
+  }
+
+  $rootScope.$watch(session.getToken, function () {
+    $http.defaults.headers.common.Token = session.token;
+  });
+
+  return session;
+}).factory('profile', function ($resource, serverConfig, $q, iStorage, $rootScope) {
+
+  var percents = {
```

Comment:
@austinpapp When introducing Karma, we will want to assign points to filling out these fields... https://trello.com/c/iS75koMS